src/inquirer/questions.py

```diff
@@ -76,6 +77,12 @@ def add_choice(self, choice):
             self._choices.append(choice)
             return len(self._choices) - 1
 
+    def apply_filter(self, filter_func):
+        self._filtered_choices = filter_func(self.unfiltered_choices_generator)
+
+    def remove_filter(self):
+        self._filtered_choices = None
```

Comment:
these are only relevant to the new class and shouldn't be inherited by every type of question